Foundation/NSCoder.swift

```diff
@@ -126,11 +126,12 @@ public class NSCoder : NSObject {
         encodeObject(object)
     }
     
-    public func encodeArrayOfObjCType(type: UnsafePointer<Int8>, count: Int, at array: UnsafePointer<Void>) {
-        encodeValueOfObjCType("[\(count)\(String(cString: type))]", at: array)
+    public func encodeArrayOfObjCType(type: UnsafePointer<Int8>, count: Int, at array: UnsafePointer<Void>?) {
+        // FIXME: What about empty arrays at null addresses?
+        encodeValueOfObjCType("[\(count)\(String(cString: type))]", at: array!)
```

Comment:
I wasn't sure what to do in this case; this method can be called from `encodeBytes(_:length:)` with a null array. Maybe they should both have a separate helper function?